tgui/packages/tgui/interfaces/CharacterPreferences/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { useBackend, useLocalState, useSharedState } from '../../backend';
+import {
+  Box,
+  Button,
+  ByondUi,
+  Divider,
+  Flex,
+  Icon,
+  LabeledList,
+  NoticeBox,
+  Section,
+  Stack,
+  Tabs,
+} from '../../components';
+import { Window } from '../../layouts';
+import { CharacterPreferencesData, CharacterPreferencesProfile, CharacterPreferencesTabKeys } from './type';
+import { SavesTab } from './SavesTab';
+import { CharacterTab } from './CharacterTab';
+import { GeneralTab } from './GeneralTab';
+import { GameSettingsTab } from './GameSettingsTab';
+import { Fragment } from 'inferno';
+
+export const CharacterPreferences = (_props, context) => {
+  const { act, data } = useBackend<CharacterPreferencesData>(context);
+  const [menu, setMenu] = useSharedState(context, 'menu', CharacterPreferencesTabKeys.General);
```

Comment:
Assuming `useSharedState` over `useLocalState` to persist the selected tab between opens?